echo_test.go

```diff
@@ -17,11 +17,114 @@
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 const prefixURI = "/v0"
 
 func TestInvalidMethod(t *testing.T) {
 
+	req, err := http.NewRequest("GET", "/v0/echo", nil)
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+	handler := EchoHandler{}
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusMethodNotAllowed, rr.Code, "Incorrect status code returned")
+
+}
+
+func TestBadRequest(t *testing.T) {
+	msg := url.Values{}
+
+	req, err := http.NewRequest("POST", "/v0/echo", strings.NewReader(msg.Encode()))
```

Comment:
```suggestion
	req, err := http.NewRequest("POST", "/v0/echo", strings.NewReader(""))
```
It looks like `msg.Encode()` for an empty `url.Values{}` just returns an empty string.